src/arch/src/x86_64/mod.rs

```diff
@@ -131,7 +129,7 @@ pub fn configure_system(
 
     add_e820_entry(&mut params.0, 0, EBDA_START, E820_RAM)?;
 
-    let mem_end = guest_mem.end_addr();
+    let mem_end = guest_mem.last_addr();
```

Comment:
Please rename `mem_end` accordingly, it no longer represents the value where memory ends.

Reply:
Shouldn't you add 1 to `let mem_end = guest_mem.last_addr()` (line 139) and `mem_end.unchecked_offset_from(first_addr_past_32bits)` (line 157)?

Reply:
You are right. Fixed.